.github/workflows/ci.yml

```diff
@@ -0,0 +1,83 @@
+name: ci
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+
+# limit the access of the generated GITHUB_TOKEN
+permissions:
+  contents: read
+
+env:
+  DEFAULT_GO_VERSION: '1.15.10'
```

Comment:
Instead of using this pattern, I think it might be better to use go-version-file: go.mod. That will cause the use of the minimum Go minor version supported by the project, as defined in go.mod. i.e. 1.15.

WDYT? (I realise you're carrying across the default we had in Jenkinsfile. May as well improve things while we're at it :))

Reply:
> WDYT? (I realise you're carrying across the default we had in Jenkinsfile. May as well improve things while we're at it :))

Totally in

Reply:
```suggestion
```